dotnet/src/Connectors/Connectors.Memory.AzureCognitiveSearch/AzureCognitiveSearchMemoryStore.cs

```diff
@@ -428,5 +429,16 @@ private static T RunMemoryStoreOperation<T>(Func<T> operation)
         }
     }
 
+    private static double ScoreToCosineSimilarity(double score)
```

Comment:
Again, I just hope this this right because the "similarities" returned by this method are always negative for input scores above 0!

Reply:
the value turns to positive after 0.5

for score = 0.3333: `1 - (1 - 0.3333) / 0.3333` => `-1`
for score = 0.4: `1 - (1 - 0.4) / 0.4` => `-0.5`
for score = 0.5: `1 - (1 - 0.5) / 0.5` => `0`
for score = 0.6: `1 - (1 - 0.6) / 0.6` => `0.3333`
for score = 0.7: `1 - (1 - 0.7) / 0.7` => `0.57`

Reply:
For score (D) to similarity (S) aka f(D) = S ....
The domain for D is [1/3, infinity).
This makes the range for S to be [-1, 2)

S is negative for D in [1/3, 1/2) , S is 0 for D at 1/2, and S is positive for D in (1/2, infinity)

Reply:
Oh, right!  Well, if you're happy with this, I'm happy too.